functions/src/alchemy_webhook.ts

```diff
@@ -0,0 +1,44 @@
+import * as functions from "firebase-functions";
+import * as admin from "firebase-admin";
+
+// TODO: webhook receiver from alchemy
+export const alchemyWebhook = functions.https.onRequest(async (req, res) => {
+
+	// const body = req.body;
+	// const event = body.event;
+	// const alchemyWebhookId = body.webhookId;
+	// const network = event.network;
+
+	// // list of activities
+	// const activitities = event.activity;
+  res.status(200).send();
+});
+
+
+export const setRealTimeCashAddress = functions.https.onCall(async (data, context) => {
+  if (!context.auth) {
+    throw new functions.https.HttpsError("permission-denied", "missing auth");
+  }
+	// console.log("data", data);
+	const userId = data?.userId;
```

Comment:
`context.auth.uid`, we should never trust user input.

```suggestion
	const userId = context.auth.uid;
```

Reply:
https://github.com/muxable/rtchat/blob/4e4bd92591f6d172dd9c187f8eca6bd0717cc1ce/lib/models/adapters/donations.dart#L29-L32

but the function in adapter does take in a `userId` from `UserModel`. Is `userId` the same as `context.auth.uid`?

Reply:
Yes it is, actually that's a mistake, the `userId` doesn't need to be passed to that function.

https://github.com/muxable/rtchat/blob/617c57ea4b0a41aec8c10ca409cbea37e00c0bda/lib/screens/settings/donations.dart#L18